lib/src/Uncertainty/Distribution/TruncatedDistribution.cxx

```diff
@@ -227,74 +204,82 @@ Point TruncatedDistribution::getRealization() const
   for (;;)
   {
     Point realization(distribution_.getRealization());
-    if ((!finiteLowerBound_ || realization[0] >= lowerBound_) && (!finiteUpperBound_ || realization[0] <= upperBound_)) return realization;
+    if (bounds_.contains(realization)) return realization;
   }
 }
 
 /* Get the DDF of the distribution: DDF_trunc = 1[a, b] * DDF / (CDF(b) - CDF(a)) */
 Point TruncatedDistribution::computeDDF(const Point & point) const
 {
-  if (point.getDimension() != 1) throw InvalidArgumentException(HERE) << "Error: the given point must have dimension=1, here dimension=" << point.getDimension();
+  if (point.getDimension() != getDimension()) throw InvalidArgumentException(HERE) << "Error: the given point must have dimension=" << getDimension() << ", here dimension=" << point.getDimension();
 
-  const Scalar x = point[0];
-  if ((x <= lowerBound_) || (x > upperBound_)) return Point(1, 0.0);
+  if (!bounds_.contains(point)) return Point(getDimension(), 0.0);
   return normalizationFactor_ * distribution_.computeDDF(point);
 }
 
 
 /* Get the PDF of the distribution: PDF_trunc = 1[a, b] * PDF / (CDF(b) - CDF(a)) */
 Scalar TruncatedDistribution::computePDF(const Point & point) const
 {
-  if (point.getDimension() != 1) throw InvalidArgumentException(HERE) << "Error: the given point must have dimension=1, here dimension=" << point.getDimension();
+  if (point.getDimension() != getDimension()) throw InvalidArgumentException(HERE) << "Error: the given point must have dimension=" << getDimension() << ", here dimension=" << point.getDimension();
+
+  Interval offset(Point(getDimension(), -quantileEpsilon_), Point(getDimension(), quantileEpsilon_));
```

Comment:
Ouch! it will hurt performance a lot! This variable does not depend on point and is no modified within the method, so it should not be constructed at each call.